bl.js

```diff
@@ -174,6 +174,31 @@ BufferList.prototype.copy = function copy (dst, dstStart, srcStart, srcEnd) {
   return dst
 }
 
+BufferList.prototype.shallowSlice = function shallowSlice (start, end) {
+  start = start || 0
+  end = end || this.length
+
+  if(start < 0)
+    start += this.length
+  if(end < 0)
+    end += this.length
+
+  var startOffset = this._offset(start)
+  var endOffset = this._offset(end)
```

Comment:
no multiple `var` statements please, these 3 should be bunched up in one, comma-first, see `BufferList.prototype.copy` for a big example